packages/components/src/DateTimePickers/pickers/YearPicker/YearPicker.test.js

```diff
@@ -4,11 +4,89 @@ import { shallow } from 'enzyme';
 import YearPicker from './YearPicker.component';
 
 describe('YearPicker', () => {
-	it('should render a YearPicker', () => {
-		// when
-		const wrapper = shallow(<YearPicker />);
+	it('should render', () => {
+		const wrapper = shallow(<YearPicker
+			selectedYear={2012}
+		/>);
 
-		// then
 		expect(wrapper.getElement()).toMatchSnapshot();
 	});
+
+	it('should have exactly one selected year', () => {
+		const wrapper = shallow(<YearPicker
+			selectedYear={2012}
+		/>);
+
+		const actions = wrapper.find('PickerAction');
+
+		expect(actions.length).toBe(5);
+
+		const selectedActions = actions.filterWhere(action => action.prop('isSelected') === true);
+
+		expect(selectedActions.length).toBe(1);
+
+		selectedActions.forEach(action => {
+			expect(action.prop('label')).toBe('2012');
+		});
+	});
+
+	it('should callback with the year picked', () => {
+		const yearToSelect = 2013;
+		const onSelect = jest.fn();
+
+		const wrapper = shallow(<YearPicker
+			selectedYear={2012}
+			onSelect={onSelect}
+		/>);
+
+		const nextYearAction = wrapper
+			.find('PickerAction')
+			.findWhere(n => n.prop('label') === yearToSelect.toString());
+
+		expect(nextYearAction.length).toBe(1);
+
+		nextYearAction.simulate('click');
+
+		expect(onSelect).toHaveBeenCalledWith(yearToSelect);
+	});
+
+	it('should default render with current year in middle when "selectedYear" prop not provided', () => {
+		const wrapper = shallow(<YearPicker />);
+		const actions = wrapper.find('PickerAction');
+		const now = new Date();
+		const currentYear = now.getFullYear().toString();
+
+		expect(actions.at(2).prop('label')).toBe(currentYear);
+	});
+
+	it('should render with "selectedYear" prop as the middle year if provided', () => {
+		const year = 2005;
+		const wrapper = shallow(<YearPicker
+			selectedYear={year}
+		/>);
+		const actions = wrapper.find('PickerAction');
+
+		expect(actions.at(2).prop('label')).toBe(year.toString());
+	});
+
+	it('should not update the years range if the selected year change', () => {
```

Comment:
I don't understand this behavior. Shouldn't we switch to the new range containing the selected year ?

Reply:
if we do so, each time we select the new year, like the first one in top, it get automatically centered, it's really weird. For one of the next steps, the YearPicker will have full scrolling and increment/decrement functionalities. At this time,  it would have given a scroll repositioning, which is much more weirder.